fastlane/lib/fastlane/actions/set_github_release.rb

```diff
@@ -11,70 +11,83 @@ def self.run(params)
         UI.important("Creating release of #{params[:repository_name]} on tag \"#{params[:tag_name]}\" with name \"#{params[:name]}\".")
         UI.important("Will also upload assets #{params[:upload_assets]}.") if params[:upload_assets]
 
-        require 'json'
-        body_obj = {
+        repo_name = params[:repository_name]
+        api_token = params[:api_token]
+        server_url = params[:server_url]
+        tag_name = params[:tag_name]
+
+        body = {
           'tag_name' => params[:tag_name],
           'name' => params[:name],
           'body' => params[:description],
           'draft' => !!params[:is_draft],
           'prerelease' => !!params[:is_prerelease]
         }
-        body_obj['target_commitish'] = params[:commitish] if params[:commitish]
-        body = body_obj.to_json
-
-        repo_name = params[:repository_name]
-        api_token = params[:api_token]
-        server_url = params[:server_url]
-        server_url = server_url[0..-2] if server_url.end_with? '/'
+        body['target_commitish'] = params[:commitish] if params[:commitish]
 
-        # create the release
-        response = call_releases_endpoint("post", server_url, repo_name, "/releases", api_token, body)
+        GithubApiAction.run(
+          server_url: server_url,
+          api_token: api_token,
+          http_method: 'POST',
+          path: "repos/#{repo_name}/releases",
+          body: body,
+          errors: {
+            422 => proc do |result|
+              UI.error(result[:body])
+              UI.error("Release on tag #{tag_name} already exists!")
+              return nil
+            end,
+            404 => proc do |result|
+              UI.error(result[:body])
+              UI.user_error!("Repository #{repo_name} cannot be found, please double check its name and that you provided a valid API token (GITHUB_API_TOKEN)")
+            end,
+            401 => proc do |result|
```

Comment:
Same reason as https://github.com/fastlane/fastlane/pull/8971#discussion_r122365260 - let me know if you'd like me to change it. I definitely would like to, thats the whole reason for having a base class so we can respond to these generic shared errors across the board.